tests/ui/js/transformsTests.js

```diff
@@ -37,24 +37,102 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         jqUnit.assertDeepEq("Generated array values are as expected", expectedArray, tagArrayNoSpace);
     });
 
-    jqUnit.test("Test tagArrayToDisplayString function", function () {
-        jqUnit.expect(1);
+    var arrayToStringTransformTestCases = [
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {},
+            expectedResult: "Shyguy, Rootbeer"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {
+                separator: ". "
+            },
+            expectedResult: "Shyguy. Rootbeer"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {
+                separator: true
+            },
+            expectedResult: "ShyguytrueRootbeer"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {
+                separator: 1
+            },
+            expectedResult: "Shyguy1Rootbeer"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {
+                trailingSeparator: true
+            },
+            expectedResult: "Shyguy, Rootbeer, "
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer", 0, {somethingElse: ""}, ["a string in an array"]],
+            rules: {},
+            expectedResult: "Shyguy, Rootbeer"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer", 0, {somethingElse: ""}, ["a string in an array"]],
+            rules: {
+                stringOnly: false
+            },
+            expectedResult: "Shyguy, Rootbeer, 0, [object Object], a string in an array"
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer"],
+            rules: {
+                path: "innerPath"
+            },
+            expectedResult: ""
+        },
+        {
+            sourceArray: [
+                { innerPath: "Shyguy", ignoredValue: "Not a cat" },
+                { innerPath: "Rootbeer", ignoredValue: "Also not a cat" }
+            ],
+            rules: {
+                path: "innerPath"
+            },
+            expectedResult: "Shyguy, Rootbeer"
+        },
+        {
+            sourceArray: [
+                { innerPath: "Shyguy", ignoredValue: "Not a cat" },
+                { innerPath: "Rootbeer", ignoredValue: "Also not a cat" }
+            ],
+            rules: {
+                path: "notARealPath"
+            },
+            expectedResult: ""
+        },
+        {
+            sourceArray: ["Shyguy", "Rootbeer", "", false, {}, []],
+            rules: {},
+            expectedResult: "Shyguy, Rootbeer"
+        }
```

Comment:
should probably test if `sourceArray` is empty, an object ( from your implementation it looks like it would actually work), a string, undefined, and etc.